test/test_constraint.jl

```diff
@@ -105,7 +105,7 @@ function test_extension_AffExpr_scalar_constraints(
     @test c.set == MOI.EqualTo(-one(T))
     cref = @constraint(model, 2 == 1)
     c = constraint_object(cref)
-    @test isequal_canonical(c.func, zero(AffExpr))
+    @test isequal_canonical(c.func, zero(AffExprType))
```

Comment:
So breaking feature 1: constraints in extensions with no variables now return an affine expression parameterized by their variable type instead of `VariableRef`. But this is arguably a bug fix.

Reply:
Call says this is a bug fix